File: 0078-subsets/0078-subsets.java
class Solution {
    List<List<Integer>> list;
    public List<List<Integer>> subsets(int[] nums) {
        list = new ArrayList<>();
        combination(0, 0, nums, new ArrayList<>());
        return list;
    }
    private void combination(int start, int count, int[] nums, List<Integer> selected) {
        list.add(new ArrayList<>(selected));
        if(count == nums.length) {
            return;
        }
        for(int i = start; i < nums.length; i++) {
            selected.add(nums[i]);
            combination(i + 1, count + 1, nums, selected);
            selected.remove(selected.size() - 1);
        }
    }
}

File: 0078-subsets/0078-subsets.cpp
class Solution {
private:
    vector<vector<int>> answer;
    void combination(int start, int count, vector<int> nums, vector<int> selected) {
        answer.push_back(selected);
        if(count == nums.size()) return;
        for(int i = start; i < nums.size(); i++) {
            selected.push_back(nums[i]);
            combination(i + 1, count + 1, nums, selected);
            selected.pop_back();
        }
    }
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        combination(0, 0, nums, vector<int>());
        return answer;
    }
};